contracts/reputationMiningCycle/ReputationMiningCycleCommon.sol

```diff
@@ -49,6 +55,13 @@ contract ReputationMiningCycleCommon is ReputationMiningCycleStorage, PatriciaTr
       return 0;
     }
 
+    if (cachedDisputeRewardSize != 0 ) {
+      // Then we've already calculated it! Just return it
+      return cachedDisputeRewardSize;
```

Comment:
This isn't my favorite -- I'd rather have it calculated once at a known spot and then just return the storage variable, vs this memoization approach, but I can live.

Reply:
If you can point out where that spot should be, I'd gladly reconsider it. But it can't be calculated until the submission window closes, which happens at a particular time, not via a transaction. The next transaction could then be a submission claiming a bye, or one of the submissions confirming their JRH. This is the best spot that captures all those possibilities, as far as I could see.

Reply:
That makes sense... if it's a time-based condition with multiple entry points then I don't see any obvious way to go about it.